pingcap/tidb
# infoschema: Add new columns for information_schema.tiflash_tables

Signed-off-by: JaySon-Huang <tshent@qq.com>



### What problem does this PR solve?

Issue Number: related to https://github.com/pingcap/tics/pull/2229

Problem Summary: Suspicious that some TiFlash tasks hold snapshot for a long time, prevent PageStorage from GCing old PageFiles. We add some information about the oldest snapshot in TiFlash system tables and want to access them by TiDB system table `information_schema.tiflash_tables`.

### What is changed and how it works?

Add column `STORAGE_{STABLE,DELTA,META}_OLDEST_SNAPSHOT_LIFETIME` and `STORAGE_{STABLE,DELTA,META}_OLDEST_SNAPSHOT_THREAD_ID` for `information_schema.tiflash_tables`

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch: 4.0, 5.0, 5.1

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)
  - Deploy a TiFlash contains that commit
  - Deploy a TiDB contains this commit
  - Check the query result of `select * from information_schema.tiflash_tables limit 1`

Side effects

- Breaking backward compatibility

### Release note 

- No release note
